### ISSUE TITLE
NFO updater issues

### ISSUE THREAD

**Issue by C4Wiz** [b-jesch/service.nfo.updater#4]
this plugin used to have settings, now it does not.

it also seens to have a issue with sets.

```
2024-06-29 11:41:43.170 T:13618    info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
2024-06-29 11:44:47.520 T:13802    info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
2024-06-29 13:42:03.716 T:3963    error <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard update of NFO: item
2024-06-29 15:30:47.380 T:3963    error <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard update of NFO: item
2024-06-29 17:30:36.554 T:3963    error <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard update of NFO: item
2024-06-29 17:30:51.454 T:30812    info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
2024-06-29 17:33:55.704 T:30763   error <general>: CCurlFile::Open - <https://repo.kodinerds.net/addons/matrix/zip/service.sleepy.watchdog/resources/media/fanart.png> Failed with code 404:
2024-06-29 17:35:27.690 T:31147    info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
2024-06-29 18:01:58.012 T:32572   debug <general>: CPythonInvoker(212):  videolibrary.showunwatchedplots
2024-06-29 18:01:58.074 T:32572    info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
2024-06-29 18:02:23.143 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Notification received: VideoLibrary.OnUpdate - {"item":{"id":156,"type":"set"},"playcount":1}
2024-06-29 18:02:23.143 T:32516   error <general>: [Watchedstate NFO Updater 1.0.9+matrix] Could not determine media type: set
2024-06-29 18:02:23.143 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Notification received: VideoLibrary.OnUpdate - {"item":{"id":2703,"type":"movie"},"playcount":1}
2024-06-29 18:02:23.372 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] NFO nfs://10.172.69.5/mnt/user/Video/Movies/American Pie (1999)-1080p/American Pie (1999)_AR185.nfo written.
2024-06-29 18:02:27.468 T:3861    debug <general>: CDirectoryProvider[special://skin/playlists/unwatched_movies.xsp]: refreshing..
2024-06-29 18:02:27.470 T:3861    debug <general>: CDirectoryProvider[special://skin/playlists/unwatched_tvshows.xsp]: refreshing..
2024-06-29 18:02:27.546 T:32605   debug <general>: CPythonInvoker(216):  videolibrary.showunwatchedplots
2024-06-29 18:02:27.632 T:32605    info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
2024-06-29 18:03:33.834 T:32692   debug <general>: CPythonInvoker(220):  videolibrary.showunwatchedplots
2024-06-29 18:03:33.905 T:32692    info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
2024-06-29 18:03:54.846 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Playlist.OnClear
2024-06-29 18:03:54.846 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Playlist.OnAdd
2024-06-29 18:03:54.946 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Info.OnChanged
2024-06-29 18:03:54.946 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnPlay
2024-06-29 18:03:55.328 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:55.428 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:55.429 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVStart
2024-06-29 18:03:55.429 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:57.032 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:57.333 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:57.634 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:57.835 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:58.235 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:58.536 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:58.839 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:03:59.240 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:04:00.041 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnAVChange
2024-06-29 18:04:12.764 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnSpeedChanged
2024-06-29 18:04:13.866 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnSpeedChanged
2024-06-29 18:04:14.167 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnSpeedChanged
2024-06-29 18:04:14.668 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnSpeedChanged
2024-06-29 18:04:31.675 T:32700   debug <general>: DoWork - Marking video item nfs://10.172.69.5/mnt/user/Video/Movies/American Pie Presents Band Camp (2005)-1080p/American Pie Presents Band Camp (2005)_AR178.mkv as watched
2024-06-29 18:04:31.704 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Notification received: VideoLibrary.OnUpdate - {"item":{"id":2700,"type":"movie"},"playcount":1}
2024-06-29 18:04:31.863 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] NFO nfs://10.172.69.5/mnt/user/Video/Movies/American Pie Presents Band Camp (2005)-1080p/American Pie Presents Band Camp (2005)_AR178.nfo written.
2024-06-29 18:04:31.864 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Notification received: VideoLibrary.OnUpdate - {"id":2700,"type":"movie"}
2024-06-29 18:04:31.864 T:32516   error <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard update of NFO: item
2024-06-29 18:04:31.864 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Discard: Player.OnStop
2024-06-29 18:04:37.640 T:3861    debug <general>: CDirectoryProvider[special://skin/playlists/unwatched_movies.xsp]: refreshing..
2024-06-29 18:04:37.641 T:3861    debug <general>: CDirectoryProvider[special://skin/playlists/unwatched_tvshows.xsp]: refreshing..
2024-06-29 18:04:37.732 T:479     debug <general>: CPythonInvoker(230):  videolibrary.showunwatchedplots
2024-06-29 18:04:37.822 T:479      info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
2024-06-29 18:05:17.671 T:512      info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
                                                       <importwatchedstate>true</importwatchedstate>
2024-06-26 17:19:21.423 T:3861     info <general>: CAddonMgr::FindAddons: script.module.watchdog v0.10.2+matrix.1 installed
2024-06-26 17:19:21.423 T:3861     info <general>: CAddonMgr::FindAddons: service.librarywatchdog v1.1.2 installed
2024-06-26 17:19:24.600 T:3949     info <general>: set Property 'videolibrary.showunwatchedplots' to [0, 1, 2]
```

**Comment by b-jesch** [comment]
This is a service addon watching the video library activities via xbmc.Monitor(). It doesn't need a setting module.

```2024-06-29 18:02:23.143 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Notification received: VideoLibrary.OnUpdate - {"item":{"id":156,"type":"set"},"playcount":1}
2024-06-29 18:02:23.143 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Notification received: VideoLibrary.OnUpdate - {"item":{"id":156,"type":"set"},"playcount":1}
2024-06-29 18:02:23.143 T:32516   error <general>: [Watchedstate NFO Updater 1.0.9+matrix] Could not determine media type: set
2024-06-29 18:02:23.143 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] Notification received: VideoLibrary.OnUpdate - {"item":{"id":2703,"type":"movie"},"playcount":1}
2024-06-29 18:02:23.372 T:32516   debug <general>: [Watchedstate NFO Updater 1.0.9+matrix] NFO nfs://10.172.69.5/mnt/user/Video/Movies/American Pie (1999)-1080p/American Pie (1999)_AR185.nfo written.
```
If the movie with ID 2703 is part of the set with the ID 156, so all is ok. It should not fire an error message indeed.

**Comment by b-jesch** [comment]
fixed by commit [#d5e55ed](https://github.com/b-jesch/service.nfo.updater/commit/d5e55ed2b5a90f0e5017d3375cbf360101cc7060)
